src/mavsdk/core/fs_test.cpp

```diff
@@ -0,0 +1,80 @@
+#include <gtest/gtest.h>
+#include <unistd.h>
+#include "fs.h"
+
+#if defined(WINDOWS)
+#else
+#define PATH_MAX 4096
```

Comment:
This is committed, but I'm not resolving this conversation yet because what you've said deserves discussion.

So, yes, it is true that I left Windows out of these tests. The reasons are several.

First, I have no way to test Windows locally. I don't really want to rely on a slow CI build system to see if my assumptions about an OS I do not use are correct.

Second, Windows tests should almost certainly use `\` for the path separator. I know that if you're programming in python Windows does accept paths using `/`, but I have no idea how deep this support goes. It's probably unwise to rely on it.

Third, most of these tests run with directories beginning with `/`, simply to free myself of the requirement to get the CWD when I don't have to. I have no clue how Windows handles paths beginning with a `/`, especially given that an absolute path in Windows begins with something looking like `C:\`. To put it simply, many of the tests that I've written are inapplicable to non-Unix-like systems.

For these reasons, I think it would actually be best to implement an alternate test suite for Windows that can test more Windows-specific things, but I won't be able to write those tests because, again, I do not have access to a Windows machine to run them on.

Reply:
I've just realized that even if Windows accepts `/` as a path separator, it doesn't matter since `fs_canonical` only accepts the path separator.

On a related note, should we be using `std::filesystem::preferred_separator`? Right now, the preferred separator is stored as `mavsdk::path_separator` (defined in `fs.h`, if I'm not mistaken).

Reply:
> On a related note, should we be using std::filesystem::preferred_separator?

If I remember correctly, we have this filesystem implementation because `std::filesystem` was not available on all the platforms we support. Not completely sure though :sweat_smile:

Reply:
> First, I have no way to test Windows locally.

I just realized we don't run the Windows tests in the CI anyway. I would suggest we remove the `ifdef` and let the unit tests run on Windows, but you don't have to test that they work. So that if we decide to run the tests on the CI some day, they will run (and we'll fix them if they fail). My concern is that otherwise we will just forget this `ifdef` and those tests will never, ever run on Windows.

What do you think?

Reply:
I guess it's fine for now. The server is likely not on Windows anyway.